src/renderer/routes/flow_chart/views/ControlBar/ExportResultButton.tsx

```diff
@@ -3,14 +3,13 @@ import { useSocket } from "@/renderer/hooks/useSocket";
 import saveAs from "file-saver";
 
 export const ExportResultButton = () => {
-  const { states } = useSocket();
-  const { programResults } = states;
+  const { blockResults } = useSocket();
 
-  const exportResultDisabled = programResults.length == 0;
+  const exportResultDisabled = Object.keys(blockResults).length == 0;
 
   const downloadResult = async () => {
-    if (!programResults.length) return;
-    const json = JSON.stringify(programResults, null, 2);
+    if (!blockResults.length) return;
```

Comment:
`blockResults` is an Object hence `!.length` check will always return `true`.
```suggestion
    if (Object.keys(blockResults).length === 0) return;
```